accumulator_updater/programs/accumulator_updater/src/state/accumulator_input.rs

```diff
@@ -14,26 +15,105 @@ use {
 /// the CPI calling program (e.g. Pyth Oracle)
 ///
 /// TODO: implement custom serialization & set alignment
-#[account]
+#[account(zero_copy)]
+#[derive(Debug, InitSpace)]
 pub struct AccumulatorInput {
     pub header: AccumulatorHeader,
-    pub data:   Vec<u8>,
+    // 10KB - 8 (discriminator) - 2053 (header) - 11 (alignment)
+    pub data:   [u8; 8_176],
+}
+
+//TODO:
+// - implement custom serialization & set alignment
+// - what other fields are needed?
+#[zero_copy]
+#[derive(InitSpace, Debug)]
+pub struct AccumulatorHeader {
+    pub bump:     u8,                // 1
+    pub version:  u8,                // 2
+    pub unused_0: u16,               // 2
+    pub indexes:  [InputIndex; 255], // 2048
+}
+
+impl AccumulatorHeader {
+    pub const CURRENT_VERSION: u8 = 1;
+
+    pub fn new(bump: u8) -> Self {
+        Self {
+            bump,
+            unused_0: 0,
+            version: Self::CURRENT_VERSION,
+            indexes: [InputIndex {
+                offset:   0,
+                unused_0: 0,
+                len:      0,
+            }; u8::MAX as usize],
+        }
+    }
+
+    pub fn set_version(&mut self) {
+        self.version = Self::CURRENT_VERSION;
+    }
+}
+
+#[zero_copy]
+#[derive(Debug, InitSpace)]
+pub struct InputIndex {
+    pub offset:   u32,
+    pub len:      u16,
+    pub unused_0: u16,
```

Comment:
having both offset and len seems redundant? I think an `[u16]` would suffice for the `indexes` field above, where the assumptions are that the indexes occur in sorted order so you can subtract adjacent ones to get the length. e.g., if the array was like [0, 10, 12] you would have a 10 byte message at 0, then a 2 byte message at 10. 

(this would also reserve more space for actual messages, which would be good. Not great that 20% of the space goes to indexes right now)